client/query.go

```diff
@@ -70,14 +46,112 @@ func (c *QueryClient) Validate() error {
 }
 
 // QueryEvents initiates an Insights query, returns a response for parsing
-func (c *QueryClient) QueryEvents(nrqlQuery string) (*QueryResponse, error) {
+func (c *QueryClient) QueryEvents(nrqlQuery string) (response *QueryResponse, err error) {
+	response = &QueryResponse{}
+	err = c.Query(nrqlQuery, response)
+	if err != nil {
+		return nil, err
+	}
+
+	return response, nil
+}
+
+// Query initiates an Insights query, with the JSON parsed into 'response'
+func (c *QueryClient) Query(nrqlQuery string, response interface{}) (err error) {
+	if response == nil {
+		return errors.New("go-insights: Invalid query response can not be nil")
+	}
+
+	err = c.generateQueryURL(nrqlQuery)
+	if err != nil {
+		return err
+	}
+
 	c.Logger.Debugf("Querying: %s", nrqlQuery)
-	c.generateQueryURL(nrqlQuery)
+	err = c.queryRequest(response)
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
+
+// queryRequest makes a NRQL query
+func (c *QueryClient) queryRequest(queryResult interface{}) (err error) {
```

Comment:
`queryResult` or `urlResult`? I'm not sure... query result makes me think I'm passing another query's result in...

Reply:
`queryResult` is where the result of this request should go (pointer to a struct, using an `interface{}` here so that the struct can change, as long as it is decorated for the JSON library to decode into.  Perhaps better comments / docs are required here.